### PR TITLE
use call_actor callback for better reusability of library

### PR DESCRIPTION
**Pull request by stevelr**
Most of the functionality of the httpserver capability provider is wrapped in a library wasmcloud-provider-httpserver, which was intended to provide a reusable, embeddable http server library.

This change improves the reusability of the library by allowing customization of `call_actor`: the asynchronous function that delivers the http payload to an actor, waits for the result, and returns the response to the http caller.

With this change, the http server can use different transports to communicate with the actor, for example, by sending the http payload over nats, instead of wasmbus-rpc.

The wasmcloud-provider-httpserver library has not yet been published to crates.io. We'd like to publish it as a crate after this PR is merged.

### CODE REVIEW

**Comment by stevelr** [comment]
Dependent on https://github.com/wasmCloud/wasmcloud-test/pull/24, which is in turn dependent on https://github.com/wasmCloud/weld/pull/143

**Comment by stevelr** [comment]
-  For safety, sets a maximum size of incoming content to 100MB. The maximum can be adjusted anywhere from 0 bytes to 1 GB by setting `max_content_len` in the configuration settings or in the linkdef values.

**Comment by stevelr** [comment]
Other than references in Cargo.toml (which will be changed when dependent PRs are merged), this is ready for review

**Comment by stevelr** [comment]
blocked on https://github.com/wasmCloud/wasmcloud-test/pull/24

File: httpserver-rs/Cargo.toml
Comment (brooksmtownsend) [inline httpserver-rs/Cargo.toml:33]: reminder to return to version after that PR merges

File: httpserver-rs/bin/main.rs
Comment (brooksmtownsend) [inline httpserver-rs/bin/main.rs:86]: this looks like it should be commented in

File: httpserver-rs/bin/main.rs
Comment (brooksmtownsend) [inline httpserver-rs/bin/main.rs:88]: If lattice ID isn't needed here, when would it be useful? I figured that the httpserver couldn't send to anything but the lattice that it belongs to anyways

File: httpserver-rs/src/settings.rs
Comment (brooksmtownsend) [inline httpserver-rs/src/settings.rs:45]: Would you mind adding this into the README of this provider so that we know the syntax for overriding the linkdef variable?

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]

File: httpserver-rs/bin/main.rs
Comment (stevelr) [reply httpserver-rs/bin/main.rs:88]: It's needed if the host & capability provider are "multi-tenant"/multi-lattice.

File: httpserver-rs/Cargo.toml
Comment (stevelr) [reply httpserver-rs/Cargo.toml:33]: 👍 still waiting

**Review by brooksmtownsend — COMMENTED** [review]
LGTM once git dependencies are pinned dependencies

File: httpserver-rs/Cargo.toml
Comment (stevelr) [reply httpserver-rs/Cargo.toml]: done!

**Comment by stevelr** [comment]
Note: after this PR is merged, tag and release wasmcloud-httpserver-provider library as a new crate

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**
